src/python/WMCore/Storage/StageOutImpl.py

```diff
@@ -236,6 +267,9 @@ def __call__(self, protocol, inputPFN, targetPFN, options=None, checksums=None):
         # This block will now always be executed after retries are exhausted
         if stageOutEx is not None:
             logging.error("Maximum number of retries exhausted. Further details on the failed command reported below.")
-            command = self.createDebuggingCommand(sourcePFN, targetPFN, options, checksums)
+            try:
+                command = self.createDebuggingCommand(sourcePFN, targetPFN, options, checksums, authmethod="TOKEN")
+            except TypeError as ex:
+                command = self.createDebuggingCommand(sourcePFN, targetPFN, options, checksums)
```

Comment:
This construct is a little bit foggy to me. I guess it is trying to create the debugging command supposing default auth method as `TOKEN` and if it fails, only then it tries the alternative. If it is so, then we may get ourselves trapped in the future, in case the line `command = self.createDebuggingCommand(sourcePFN, targetPFN, options, checksums, authmethod="TOKEN")` throws a `TypeError` exception of origin different than auth method mismatch.  The better approach to me would look like creating  a dedicated Exception for that and check for it explicitly. 

Are  we always identifying the authType this way?

Reply:
Hi @todor-ivanov , I've added this try-except to manage the cases where a submodule does not support `authmethod` yet, as for now we are supposed to support tokens only for gfal and xrootd. `TypeError` is the type of exception returned if a submodule different from `GFAL2Impl` or `XRDImpl` is used. Do you still think we need a dedicated exception to be introduced?

Reply:
> TypeError is the type of exception returned if a submodule different from GFAL2Impl or XRDImpl is used.

Well, that's the risky part. Because `TypeError` is a generic python error, this could be thrown due to many reasons from those modules. And distinguishing the origin based on generic exception is risky. 

> Do you still think we need a dedicated exception to be introduced?

This is a tricky question. My 6th sense tells me we should. While, on the other hand, I know how  convoluted the stage out module is, so finding all places where we could break the call due to an unsupported authentication method, could be too much work. So it is your call. If you think it would be easy to switch to a dedicated exception, that would be good. But we can also give it a try as it is.

Reply:
If all the method signatures have been updated, should we ditch off the try/except block?